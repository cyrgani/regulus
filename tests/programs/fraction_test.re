import(fraction),

=(x, Fraction(2, 1)),
assert_eq(fraction_to_int(x), 2),
assert_eq(fraction_to_int(Fraction(3, 2)), 1),
assert_eq(fraction_to_int(Fraction(5, 2)), 2),
assert_eq(fraction_to_int(Fraction(-3, 2)), -1),
assert_eq(fraction_to_int(Fraction(-1, 2)), 0),
assert_eq(catch(fraction_to_int(Fraction(0, 0))), "OverflowError: overflow occured during /!"),
