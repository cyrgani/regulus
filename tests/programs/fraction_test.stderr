DivideByZeroError: attempted to divide by zero
at <stl:operators>:55:26
at <stl:operators>:53:11
at <stl:operators>:50:19
at <stl:fraction>:20:6
at <stl:fraction>:19:25
at programs/fraction_test.re:9:34
at programs/fraction_test.re:9:22
